code/modules/power/supermatter/supermatter_cascade_components.dm

```diff
@@ -0,0 +1,256 @@
+		return PROCESS_KILL
+
+	COOLDOWN_START(src, sm_wall_cooldown, rand(0, 5 SECONDS))
+
+	var/picked_dir = pick_n_take(available_dirs)
+	var/turf/next_turf = get_step_multiz(src, picked_dir)
+	if(!istype(next_turf) || istype(next_turf, /turf/closed/indestructible/supermatter_wall))
+		return
+
+	icon_state = "crystal_cascade_[rand(1,6)]"
+
+	for(var/atom/movable/checked_atom as anything in next_turf)
+		if(istype(checked_atom, /mob/living))
+			qdel(checked_atom)
+		else if(istype(checked_atom, /mob)) // Observers, AI cameras.
+			continue
+		else if(istype(checked_atom, /obj/cascade_portal))
+			continue
+		else
+			qdel(checked_atom)
+		CHECK_TICK
+
+	next_turf.ChangeTurf(type)
+	var/turf/closed/indestructible/supermatter_wall/sm_wall = next_turf
+	if(sm_wall.available_dirs)
+		sm_wall.available_dirs -= get_dir(next_turf, src)
+
+/turf/closed/indestructible/supermatter_wall/bullet_act(obj/projectile/projectile)
+	visible_message(span_notice("[src] is unscathed!"))
+	return BULLET_ACT_HIT
+
+/turf/closed/indestructible/supermatter_wall/singularity_act()
+	return
+
+/turf/closed/indestructible/supermatter_wall/blob_act(obj/structure/blob/blob)
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, TRUE)
+	Consume(blob)
+
+/turf/closed/indestructible/supermatter_wall/attack_tk(mob/user)
+	if(!iscarbon(user))
+		return
+	var/mob/living/carbon/jedi = user
+	to_chat(jedi, span_userdanger("That was a really dense idea."))
+	jedi.ghostize()
+	var/obj/item/organ/brain/rip_u = locate(/obj/item/organ/brain) in jedi.internal_organs
+	if(rip_u)
+		rip_u.Remove(jedi)
+		qdel(rip_u)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/turf/closed/indestructible/supermatter_wall/attack_paw(mob/user, list/modifiers)
+	dust_mob(user, cause = "monkey attack")
+
+/turf/closed/indestructible/supermatter_wall/attack_alien(mob/user, list/modifiers)
+	dust_mob(user, cause = "alien attack")
+
+/turf/closed/indestructible/supermatter_wall/attack_animal(mob/living/simple_animal/user, list/modifiers)
+	var/murder
+	if(!user.melee_damage_upper && !user.melee_damage_lower)
+		murder = user.friendly_verb_continuous
+	else
+		murder = user.attack_verb_continuous
+	dust_mob(user, \
+	span_danger("[user] unwisely [murder] [src], and [user.p_their()] body burns brilliantly before flashing into ash!"), \
+	span_userdanger("You unwisely touch [src], and your vision glows brightly as your body crumbles to dust. Oops."), \
+	"simple animal attack")
+
+/turf/closed/indestructible/supermatter_wall/attack_robot(mob/user)
+	if(Adjacent(user))
+		dust_mob(user, cause = "cyborg attack")
+
+/turf/closed/indestructible/supermatter_wall/attack_ai(mob/user)
+	return
+
+/turf/closed/indestructible/supermatter_wall/attack_hulk(mob/user)
+	dust_mob(user, cause = "hulk attack")
+
+/turf/closed/indestructible/supermatter_wall/attack_larva(mob/user)
+	dust_mob(user, cause = "larva attack")
+
+/turf/closed/indestructible/supermatter_wall/attack_hand(mob/living/user, list/modifiers)
```

Comment:
Mildly cringe that literally all this supermatter stuff is copy+pasted directly from the supermatter itself.

Reply:
i agree

Reply:
Yeah but I don't know how to not do that

Reply:
Don't we have a supermatter component? If so, just apply that to the walls and make sure the effects are extended to turfs.

Otherwise, make a component to extend that behavior.